rust/scx_rustland_core/assets/bpf/main.bpf.c

```diff
@@ -188,6 +200,13 @@ struct {
 	__type(value, struct usersched_timer);
 } usersched_timer SEC(".maps");
 
+/*
+ * Time period of the scheduler heartbeat, used to periodically kick the the
+ * scheduler and check if we need to switch to FIFO mode or regular
+ * scheduling.
+ */
+#define USERSCHED_TIMER_NS NSEC_PER_SEC
```

Comment:
It's just my intuition but if the overhead isn't too high, it might be worthwhile to try to keep this duration something which people usually don't notice - say. something under 100ms. Something which is best decided with experiments, probably.

Reply:
Yeah, I was also thinking about this, the overhead shouldn't be that much, we're just accessing mostly local variables in the timer callback.

I think a potential issue of reducing the timer too much could be that we may get a lot of false positives and we may start jumping back and forth into FIFO mode. In this case we may want to refine the moving average to better "smooth" the potential false positives. Because ideally we want to act promptly at exiting from FIFO mode, while entering in FIFO mode is not so critical.

I'll run more tests trying different values of `USERSCHED_TIMER_NS`, potentially also with a longer time window for the `nr_active_avg` moving average.